src/main/java/tutorly/logic/commands/EditSessionCommand.java

```diff
@@ -0,0 +1,175 @@
+
+    public static final String MESSAGE_USAGE = COMMAND_STRING
+            + ": Edits the details of the session identified by a SESSION_ID. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: SESSION_ID "
+            + PREFIX_SESSION + "SESSION_ID\n"
+            + "[" + PREFIX_TIMESLOT + "TIMESLOT] "
+            + "[" + PREFIX_SUBJECT + "SUBJECT] "
+            + "Example: " + COMMAND_STRING + " 1 "
+            + PREFIX_TIMESLOT + "30 Mar 2025 11:30-13:30 "
+            + PREFIX_SUBJECT + "Mathematics";
+
+    public static final String MESSAGE_EDIT_SESSION_SUCCESS = "Edited session: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final int sessionId;
+    private final EditSessionDescriptor editSessionDescriptor;
+
+    /**
+     * @param id containing the ID or name of the session to edit
+     * @param editSessionDescriptor details to edit the session with
+     */
+    public EditSessionCommand(int id, EditSessionDescriptor editSessionDescriptor) {
+        requireNonNull(editSessionDescriptor);
+
+        this.sessionId = id;
+        this.editSessionDescriptor = new EditSessionDescriptor(editSessionDescriptor);
+    }
+
+    /**
+     * Creates and returns a {@code Session} with the details of {@code sessionToEdit}
+     * edited with {@code editSessionDescriptor}. The ID of the session cannot be edited.
+     */
+    private static Session createEditedSession(Session sessionToEdit, EditSessionDescriptor editSessionDescriptor) {
+        assert sessionToEdit != null;
+
+        Timeslot updatedDate = editSessionDescriptor.getTimeslot().orElse(sessionToEdit.getTimeslot());
+        String updatedSubject = editSessionDescriptor.getSubject().orElse(sessionToEdit.getSubject());
+
+        Session newSession = new Session(updatedDate, updatedSubject);
+        newSession.setId(sessionToEdit.getId());
+
+        return newSession;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Optional<Session> sessionToEdit = model.getSessionById(sessionId);
+        if (sessionToEdit.isEmpty()) {
+            throw new CommandException(Messages.MESSAGE_SESSION_NOT_FOUND);
+        }
+
+        Session editedSession = createEditedSession(sessionToEdit.get(), editSessionDescriptor);
+
+        if (model.hasOverlappingSession(editedSession)) {
+            throw new CommandException(Messages.MESSAGE_SESSION_OVERLAP);
+        }
+
+        model.setSession(sessionToEdit.get(), editedSession);
+        model.updateFilteredSessionList(FILTER_SHOW_ALL_SESSIONS);
+
+        return new CommandResult.Builder(String.format(MESSAGE_EDIT_SESSION_SUCCESS, Messages.format(editedSession)))
+                .withTab(Tab.SESSION)
+                .build();
```

Comment:
There's a missing reverse command here
```suggestion
        return new CommandResult.Builder(String.format(MESSAGE_EDIT_SESSION_SUCCESS, Messages.format(editedSession)))
                .withTab(Tab.SESSION)
                .withReverseCommand(new EditSessionCommand(
                        sessionId, EditSessionDescriptor.fromSession(sessionToEdit.get())))
                .build();
```